server/index.js

```diff
@@ -1,24 +1,127 @@
+/* eslint-disable guard-for-in */
+/* eslint-disable camelcase */
+/* eslint-disable no-console */
+/* eslint-disable no-restricted-syntax */
 const express = require('express');
-const db = require('./database/db.js');
 const path = require('path');
+const moment = require('moment');
+const db = require('./database/db.js');
 
 const app = express();
 
+app.use((req, res, next) => {
+  console.log(`serving ${req.method} for ${req.url}`);
+  next();
+});
+
 app.use(express.json());
 app.use('/hostels/:hostel_id', express.static(path.join(__dirname, '../public')));
 
-
 app.get('/hostels/:id/api/reviews', (req, res) => {
   const queryStr = `SELECT * FROM reviews INNER JOIN authors ON reviews.author_id = authors.id WHERE reviews.hostel_id = ${req.params.id}`;
   db.connection.query(queryStr, (err, response) => {
     if (err) {
-      console.log(err);
+      console.log(`an error occured getting all reviews for hostel id ${req.params.id}`, err);
       res.sendStatus(500);
     } else {
-      console.log('done');
+      console.log('query successful');
       res.json(response);
     }
   });
 });
 
+app.get('/api/reviews/:id', (req, res) => {
+  const queryStr = `SELECT * FROM reviews INNER JOIN authors ON reviews.author_id = authors.id WHERE reviews.id = ${req.params.id}`;
+  db.connection.query(queryStr, (err, response) => {
+    if (err) {
+      console.log(`an error occured getting review id ${req.params.id}`, err);
+      res.sendStatus(500);
+    } else {
+      console.log('query successful');
+      res.json(response);
+    }
+  });
+});
+
+app.post('/api/reviews', (req, res) => {
+  if (req.body === undefined) {
+    res.status(400).send('please include data to populate the review');
+  }
+  const {
+    hostel_id, author_id, description, security, location,
+    staff, atmosphere, cleanliness, facilities, value, total,
+  } = req.body;
+
+  // check if any fields are missing
+  if ((!hostel_id || !author_id || !description || !security
+    || !location || !staff || !atmosphere || !cleanliness
+    || !facilities || !value || !total) === undefined) {
+    res.status(400).send(`the following fields are required: {
+      hostel_id: <Number>,
+      author_id: <Number>,
+      description: <String>, // max 255 characters
+      security: <Number>, // rating out of 10
+      location: <Number>, // rating out of 10
+      staff <Number>, // rating out of 10
+      atmosphere <Number>, // rating out of 10
+      cleanliness <Number>, // rating out of 10
+      facilities <Number>, // rating out of 10
+      value <Number>, // rating out of 10
+      total <Number>, // average of all ratings,
```

Comment:
This works, but it looks messy. I wonder if you could somehow use `object.hasOwnProperty( prop )`?

Reply:
What do you mean? Would I be checking req.body.hasOwnProperty(prop)?

Reply:
Nice catch. Implemented (just for the practice).